apis/eventsApi.js

```diff
@@ -0,0 +1,26 @@
+const express = require('express');
+const router = express.Router();
+const generateMockEvents = require('./mock/mockEventsApi');
+
+var events = [];
+
+events = generateMockEvents();
+
+// Get events with the given room and events in which the given time falls in between the event's start and end times
+function getEvents(room, time) {
+    var filteredEvents = events.filter(event => event.location == room && (time > event.startTime && event.endTime > time));
+    return filteredEvents;
+}
+
+// POST request body should be in JSON and include:
+// room: the room number to get events for
+// time: the date in milliseconds; any events happening at that date will be returned
+router.post('/getEvents', (req, res) => {
+    var room = req.body.room.toString();
```

Comment:
Also two thoughts here – in the end, will we be querying scheduler once a minute for every room? Why not just have one query that gives all current events? Or better yet, to avoid being late, why not just have a query where we just get all the events that are either in progress or are slated to come up? That way, we don't need to call the endpoint so often in the first place.

Reply:
I'm going to meet with someone on the Scheduler team to discuss this. Their backend is almost done, so I'll see how it's designed and get a better idea of how to go about this